app/src/main/java/com/pmatuki/wowpapers/view/list/ListActivity.kt

```diff
@@ -29,7 +33,12 @@ class ListActivity : AppCompatActivity(), WallpaperItemClickListener {
     }
 
     private fun bindViewModel() {
-        viewModel = ViewModelProvider(this).get(WallpaperListViewModel::class.java)
+        viewModel = ViewModelProvider(
+            this, WallpaperListViewModelFactory(
+                WallpaperDataSource(WallhavenService()), WallpaperMapper()
```

Comment:
I removed DI from ViewModels and I'm passing dependencies in the constructor.

I don't like that ListActivity has to know so much about how to instance a WallpaperDataSource. It also has to reference WallhavenService and WallpaperMapper, which feels wrong.

@igorescodro : should I inject the dependencies inside the Factory class instead? After all, shouldn't a factory class  know how to build the object it is meant to provide without the Activity having to specify its dependencies?